pallets/did/src/origin.rs

```diff
@@ -36,19 +37,43 @@ impl<DidIdentifier, AccountId> DidRawOrigin<DidIdentifier, AccountId> {
 	}
 }
 
-pub struct EnsureDidOrigin<DidIdentifier, AccountId>(PhantomData<(DidIdentifier, AccountId)>);
+impl<DidIdentifier: Clone, AccountId: Clone> CallSources<AccountId, DidIdentifier>
+	for DidRawOrigin<DidIdentifier, AccountId>
+{
+	fn sender(&self) -> AccountId {
+		self.submitter.clone()
+	}
 
-impl<OuterOrigin, DidIdentifier, AccountId> EnsureOrigin<OuterOrigin> for EnsureDidOrigin<DidIdentifier, AccountId>
+	fn subject(&self) -> DidIdentifier {
+		self.id.clone()
+	}
+}
+
+pub struct EnsureDidOrigin<DidIdentifier, AccountId, ExpectedSubmitter = ()>(
```

Comment:
Its more `EnsureDidAuthorizedOrigin`. Feel free to resolve this.

Reply:
Yeah in the beginning I went for a new struct altogether, but then realised it's actually easier to change the existing one, so `EnsureDidOrigin` is the origin check we already have in place, and I don't think renaming it right now is a good idea.